src/main/java/io/jenkins/plugins/checks/github/GitHubSCMSourceChecksContext.java

```diff
@@ -20,100 +17,118 @@
+        runAvailable = false;
     }
 
     @Override
     public String getHeadSha() {
-        return resolveHeadSha();
+        String headSha = resolveHeadSha();
+        if (StringUtils.isBlank(headSha)) {
+            throw new IllegalStateException("No SHA found for job: " + getJob().getName());
+        }
+
+        return headSha;
     }
 
     @Override
     public String getRepository() {
         GitHubSCMSource source = resolveSource();
-        return source.getRepoOwner() + "/" + source.getRepository();
+        if (source == null) {
+            throw new IllegalStateException("No GitHub SCM source found for job: " + getJob().getName());
+        }
+        else {
+            return source.getRepoOwner() + "/" + source.getRepository();
+        }
     }
 
-    @Override @Nullable
-    protected String getCredentialsId() {
-        return resolveSource().getCredentialsId();
-    }
+    @Override
+    public boolean isValid(final PluginLogger logger) {
+        if (resolveSource() == null) {
+            logger.log("No GitHub SCM source found");
 
-    private GitHubSCMSource resolveSource() {
-        Optional<GitHubSCMSource> source = getScmFacade().findGitHubSCMSource(getJob());
-        if (!source.isPresent()) {
-            throw new IllegalStateException("No GitHub SCM source available for job: " + getJob().getName());
+            return false;
+        }
+
+        if (!hasValidCredentials(logger)) {
+            return false;
         }
 
-        return source.get();
+        return StringUtils.isNotBlank(resolveHeadSha());
     }
 
-    private String resolveHeadSha() {
-        Optional<SCMHead> head = getScmFacade().findHead(getJob());
-        if (!head.isPresent()) {
-            throw new IllegalStateException("No SCM head available for job: " + getJob().getName());
+    @Override
+    @Nullable
+    protected String getCredentialsId() {
+        GitHubSCMSource source = resolveSource();
+        if (source == null) {
+            return null;
         }
 
-        Optional<SCMRevision> revision = getScmFacade().findRevision(resolveSource(), head.get());
-        if (!revision.isPresent()) {
-            throw new IllegalStateException(
-                    String.format("No SCM revision available for repository: %s and head: %s",
-                            getRepository(), head.get().getName()));
-        }
+        return source.getCredentialsId();
+    }
 
-        if (revision.get() instanceof AbstractGitSCMSource.SCMRevisionImpl) {
-            return ((AbstractGitSCMSource.SCMRevisionImpl) revision.get()).getHash();
-        }
-        else if (revision.get() instanceof PullRequestSCMRevision) {
-            return ((PullRequestSCMRevision) revision.get()).getPullHash();
+    @Nullable
+    private GitHubSCMSource resolveSource() {
+        return getScmFacade().findGitHubSCMSource(getJob()).orElse(null);
+    }
+
+    @Nullable
+    private String resolveHeadSha() {
+        if (runAvailable) {
```

Comment:
Rather than using an `if else` here and adding a new field `runAvailable` it would be nicer if you would assign the SHA directly in the corresponding `Job` and `Run` constructors.